### PR TITLE
Fixes #13247: Replace gunicorn+meinheld with uWSGI

### LINKED ISSUE #13247: Find a Replacement for Meinheld WSGI Server
## Description

Currently, our project relies on the [Meinheld](https://github.com/mopemope/meinheld) WSGI server as the underlying server technology, along with gunicorn, for serving our application. However, this project has not been updated, maintained, or seen a release in over 3 years. It also does not compile on Python versions > 3.9. 

This GitHub issue aims to discuss and identify a suitable replacement for Meinheld.

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: bin/run-prod.sh [hunk 1]
```diff
@@ -4,12 +4,11 @@
...
 
-function run-gunicorn () {
+function run-uwsgi () {
     if [[ -z "$NEW_RELIC_LICENSE_KEY" ]]; then
-        exec gunicorn "$@"
+        exec uwsgi "$@"
     else
-        export NEW_RELIC_CONFIG_FILE=newrelic.ini
-        exec newrelic-admin run-program gunicorn "$@"
+        NEW_RELIC_CONFIG_FILE=newrelic.ini newrelic-admin run-program uwsgi "$@"
```
Comment (pmac) [inline bin/run-prod.sh:11]: I think we need the exec call here so that the docker agent is attached to the right process

File: wsgi/app.py [hunk 2]
```diff
@@ -5,21 +5,18 @@
 # flake8: noqa
 import os
 
-IS_HTTPS = os.environ.get("HTTPS", "").strip() == "on"
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "bedrock.settings")
 
 # must be imported after env var is set above.
 from django.core.handlers.wsgi import WSGIRequest
```
Comment (pmac) [inline wsgi/app.py:11]: don't need this import anymore

File: bin/run-prod.sh (hunk 1 above)
Comment (robhudson) [reply bin/run-prod.sh:11]: Good point, we don't need `run-prod.sh` as the process and can replace it with the newrelic + uwsgi process. I'll update it back to the original form. I'm doing this in basket so maybe I'll update it there as well so things are the same.

Comment (robhudson) [reply bin/run-prod.sh]: Updated

**Comment by robhudson** [comment]
Updated. This is running on `www-demo2`.

**Review by pmac — APPROVED** [review]
🥇

**Pull request merged**